providers/chargebee/handlers.go

```diff
@@ -0,0 +1,100 @@
+package chargebee
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/naming"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+const (
+	apiVersion = "v2"
+)
+
+func (c *Connector) buildSingleObjectMetadataRequest(ctx context.Context, objectName string) (*http.Request, error) {
+	if objectNameWithListSuffix.Has(objectName) {
+		objectName += "/list"
+	}
+
+	url, err := urlbuilder.New(c.ProviderInfo().BaseURL, apiVersion, objectName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add limit parameter to get just a single record for sampling
+	url.WithQueryParam("limit", "1")
+
+	return http.NewRequestWithContext(ctx, http.MethodGet, url.String(), nil)
+}
+
+func (c *Connector) parseSingleObjectMetadataResponse(
+	ctx context.Context,
+	objectName string,
+	request *http.Request,
+	response *common.JSONHTTPResponse,
+) (*common.ObjectMetadata, error) {
+	objectMetadata := common.ObjectMetadata{
+		Fields:      make(map[string]common.FieldMetadata),
+		DisplayName: naming.CapitalizeFirstLetterEveryWord(objectName),
+	}
+
+	res, err := common.UnmarshalJSON[map[string]any](response)
+	if err != nil {
+		return nil, common.ErrFailedToUnmarshalBody
+	}
+
+	if res == nil || len(*res) == 0 {
```

Comment:
minor, but seems like converting to a map[string]any, and then getting an array inside it seems to be an operation that we frequently do. Might be good to add inside common/jsonquery as a helper. Doesn't block this PR, but good to do at some point.